src/diffusers/loaders/ip_adapter.py

```diff
@@ -141,12 +141,14 @@ def load_ip_adapter(
                     subfolder=os.path.join(subfolder, "image_encoder"),
                 ).to(self.device, dtype=self.dtype)
                 self.image_encoder = image_encoder
+                self.register_to_config(image_encoder=CLIPVisionModelWithProjection)
             else:
                 raise ValueError("`image_encoder` cannot be None when using IP Adapters.")
 
         # create feature extractor if it has not been registered to the pipeline yet
         if hasattr(self, "feature_extractor") and getattr(self, "feature_extractor", None) is None:
             self.feature_extractor = CLIPImageProcessor()
+            self.register_to_config(feature_extractor=["transformers", "CLIPImageProcessor"])
```

Comment:
Same. It's not supposed to be a configuration variable.

Reply:
I think it's the correct way to save the feature extractor actually